### PR TITLE
Enabled opening README.md with python2 in setup.py

### PR DESCRIPTION
**Pull request by jliersch**
In setup.py, the required python version is set to >=3.6 which would produce an error message telling someone wanting to install pyRiemann with a lower version this version mismatch. But with python < 3 the handling of this version check will not be reached. Instead setup.py fails at calling `open('README.md', 'r', encoding="utf8")`, since the parameter "encoding" was added to "open" with python3.

I propose to handle the difference in signature of "open" to provide a more precise error message if someone trys to install pyRiemann with python < 3.

### CODE REVIEW

**Comment by agramfort** [comment]
I think we should just drop the support for python < 3.6

**Comment by jliersch** [comment]
Sorry, I missed adding the corresponding issue #151.

I concur. But I also think that it should made easy for the users to find out that the python 2 support is dropped. This is usually is done by the error message about the python version mismatch (see below). But this message currently can't be reached if one tries to install pyRiemann with python < 3.

Expected error message:
`ERROR: Package 'pyriemann' requires a different Python: 2.7.18 not in '>=3.6'`

Current error message:
`ERROR: Command errored out with exit status 1:
     command: <secret path>/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-jClpfU/pyriemann/setup.py'"'"'; __file__='"'"'/tmp/pip-install-jClpfU/pyriemann/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-jClpfU/pyriemann/pip-egg-info
         cwd: /tmp/pip-install-jClpfU/pyriemann/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-jClpfU/pyriemann/setup.py", line 16, in <module>
        with open('README.md', 'r', encoding="utf8") as fid:
    TypeError: 'encoding' is an invalid keyword argument for this function
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.`

**Comment by sylvchev** [comment]
The error is easier to understand like that. Ok to merge for me.

File: setup.py [hunk 1]
```diff
@@ -13,7 +14,10 @@
 if version is None:
     raise RuntimeError("Could not determine version")
 
-with open("README.md", "r", encoding="utf8") as fid:
+# "open" does not provide the parameter "encoding" for python<3
```
Comment (qbarthelemy) [inline setup.py:17]:
```suggestion
```

**Review by qbarthelemy — APPROVED** [review]

**Comment by qbarthelemy** [comment]
Thx @jliersch !

**Pull request merged**